ImperatorToCK3/Source/CK3/Character/CK3Character.cpp

```diff
@@ -22,9 +24,22 @@ void CK3::Character::initializeFromImperator(
 	name = imperatorCharacter->getName();
 	female = imperatorCharacter->isFemale();
 	age = imperatorCharacter->getAge();
+
+	
+	unsigned long long ck3Province; // for religion mapper
+	// Determine valid (not dropped in province mappings) "source province" to be used by religion mapper. Don't give up without a fight.
+	auto impProvForProvinceMapper = imperatorCharacter->getProvince();
+	if (provinceMapper.getCK3ProvinceNumbers(impProvForProvinceMapper).empty() && imperatorCharacter->getFather().second) impProvForProvinceMapper = imperatorCharacter->getFather().second->getProvince();
+	if (provinceMapper.getCK3ProvinceNumbers(impProvForProvinceMapper).empty() && imperatorCharacter->getMother().second) impProvForProvinceMapper = imperatorCharacter->getMother().second->getProvince();
```

Comment:
else if or you're overriding best results.

Reply:
Are you sure? I intend to change `impProvForProvinceMapper` until `provinceMapper.getCK3ProvinceNumbers()` returns a non-empty set.

Reply:
pardon, you're right. You're checking empty on every one. This is fine.